pkg/cli/serve/serve.go

```diff
@@ -130,13 +186,54 @@ func (o *serveOptions) Run() error {
 		return err
 	}
 
-	// Initialize ocm sdk connection client
-	sdkclient, err := ocm.NewConnection().Build(viper.GetString(config.OcmURL),
-		viper.GetString(config.ClusterID),
-		viper.GetString(config.AccessToken))
-	if err != nil {
-		o.logger.WithError(err).Fatal("Can't initialise OCM sdk.Connection client")
-		return err
+	// Depending on whether the FleetMode is enabled or not, we need to initiate the OCM SDK connection accordingly
+	// If fleet mode is not enabled, we will fetch the cluster ID and access token to initiate connection with OCM
+	if !o.fleetMode {
```

Comment:
why do we need to talk to ocm here?

Reply:
Over here we are not really talking to OCM. We are creating the required structs which would help initiate the connection to OCM later based on the credentials passed which is dependent on the mode `ocm-agent` is running in.